components/common/hooks/useGetLimits.ts

```diff
@@ -1,31 +1,31 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {useState, useEffect, useMemo} from 'react';
+import {useEffect, useMemo} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
 
-import {Limits} from '@mattermost/types/cloud';
-import {getCloudLimits, getCloudLimitsLoaded, isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {getCloudLimits, getCloudLimitsLoaded, getSubscriptionProduct, isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
 import {getCloudLimits as getCloudLimitsAction} from 'actions/cloud';
 import {useIsLoggedIn} from 'components/global_header/hooks';
 
+import {Limits} from '@mattermost/types/cloud';
+
 export default function useGetLimits(): [Limits, boolean] {
     const isCloud = useSelector(isCurrentLicenseCloud);
     const isLoggedIn = useIsLoggedIn();
     const cloudLimits = useSelector(getCloudLimits);
+    const subscriptionProduct = useSelector(getSubscriptionProduct);
     const cloudLimitsReceived = useSelector(getCloudLimitsLoaded);
     const dispatch = useDispatch();
-    const [requestedLimits, setRequestedLimits] = useState(false);
 
     useEffect(() => {
-        if (isLoggedIn && isCloud && !requestedLimits && !cloudLimitsReceived) {
+        if (isLoggedIn && isCloud) {
             dispatch(getCloudLimitsAction());
-            setRequestedLimits(true);
         }
-    }, [isLoggedIn, isCloud, requestedLimits, cloudLimitsReceived]);
+    }, [isLoggedIn, isCloud, subscriptionProduct]);
```

Comment:
This will work for fetching subscription when the product the user is subscribed to changes, but because it's not checking whether a request needs to be made on the *first* render, we're going to have excessive calls from all the places that now always request getting limits. I propose an [alternate approach](https://github.com/mattermost/mattermost-webapp/pull/12244) where we keep the "has it already been requested" & "do we already have the data" checks, but then clear the data & the state of whether the request was made if the subscription product changes.

@AGMETEOR if that approach seems good to you, I'll add another reviewer to that PR.